htdocs/shop/creditcard_wait.bml

```diff
@@ -65,8 +65,16 @@ body<=
 
     # looks good, see what status things are in...
     $title = $ML{'.title.' . $row->{jobstate}} || $ML{'Error'};
-    return $ML{'.state.' . $row->{jobstate}}
+    my $msg = '';
+    $msg = $ML{'.state.' . $row->{jobstate}}
         if $row->{jobstate} =~ /^(?:paid|queued|internal_failure)$/;
+    # need a different message if paid but wasn't logged in
+    if ( $row->{jobstate} eq 'paid' && ! $cart->userid ) {
+        my $rcpt = LJ::create_url( '/shop/receipt',
+                                   args => { ordernum => $onum } );
```

Comment:
I had some panic about this page being open to the world... had totally forgot shop order "numbers" aren't just integers...